ttnctl/cmd/devices_convert_to_abp.go

```diff
@@ -0,0 +1,125 @@
+// Copyright © 2017 The Things Network
+// Use of this source code is governed by the MIT license that can be found in the LICENSE file.
+
+package cmd
+
+import (
+	"strings"
+
+	"github.com/TheThingsNetwork/api"
+	"github.com/TheThingsNetwork/api/handler"
+	"github.com/TheThingsNetwork/api/handler/handlerclient"
+	ttnlog "github.com/TheThingsNetwork/go-utils/log"
+	"github.com/TheThingsNetwork/ttn/core/types"
+	"github.com/TheThingsNetwork/ttn/ttnctl/util"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+func convertToABP(manager *handlerclient.ManagerClient, dev *handler.Device, flags *pflag.FlagSet) {
+	// Do all updates
+	if dev.GetLoRaWANDevice().AppKey.String() != "" {
+		old := dev.GetLoRaWANDevice().AppKey.String()
+		dev.GetLoRaWANDevice().AppKey = &types.AppKey{}
```

Comment:
The AppKey is still set, but empty. This allows anyone to activate the device with this known, empty AppKey.

Right? Or do I misunderstand how this field is interpreted?

Why not setting this field to `nil`?

Reply:
That would make sense, but it does not work like that.

Setting `AppKey` to nil here has no effect on the device. Setting it to empty, actually clears it and makes the device appear as ABP on the console.